Content.Server/GameObjects/Components/Items/RCDComponent.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using System.Collections.Generic;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Server.Interfaces;
+using Content.Shared.Maps;
+using Microsoft.EntityFrameworkCore.Internal;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components.Transform;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Map;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Map;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+
+namespace Content.Server.GameObjects.Components.Items
+{
+    [RegisterComponent]
+    public class RCDComponent : Component, IAfterAttack, IUse, IExamine
+    {
+#pragma warning disable 649
+        [Dependency] private readonly ITileDefinitionManager _tileDefinitionManager;
+        [Dependency] private readonly IEntitySystemManager _entitySystemManager;
+        [Dependency] private readonly IMapManager _mapManager;
+        [Dependency] private readonly IServerEntityManager _serverEntityManager;
+        [Dependency] private IServerNotifyManager _serverNotifyManager;
+#pragma warning restore 649
+        public override string Name => "RCD";
+        private string _outputTile = "floor_steel";
+        private RCDmode mode = 0; //What mode are we on? Can be floors, walls, deconstruct.
+        private Array modes = Enum.GetValues(typeof(RCDmode));
```

Comment:
Store this as a proper typed array instead. Untyped `Array` is generally frowned upon since you can't easily tell the type and have to do annoying casting a lot.
```suggestion
        private RCDMode[] modes = (RCDMode[]) Enum.GetValues(typeof(RCDmode));
```